docker/Dockerfile

```
@@ -1,39 +1,40 @@
-ARG ASPNET_IMAGE_TAG=8.0-bookworm-slim-amd64
-ARG NODEJS_IMAGE_TAG=22-bullseye-slim
-ARG DOTNET_SDK=8.0
+# Set the major version of dotnet
+ARG DOTNET_VERSION=8.0
+# Set the major version of nodejs
+ARG NODEJS_VERSION_MAJOR=22
 
-# Stage 1 - Build frontend assets
-FROM node:${NODEJS_IMAGE_TAG} AS assets
-COPY . /repo
-WORKDIR /repo/DfE.FindInformationAcademiesTrusts
+# Build frontend assets using node js
+FROM "node:${NODEJS_VERSION_MAJOR}-bullseye-slim" AS assets
+WORKDIR /app
+COPY ./DfE.FindInformationAcademiesTrusts/ /app
 RUN npm install
 RUN npm run build
 
-# Stage 2 - Build and publish dotnet application
-FROM mcr.microsoft.com/dotnet/sdk:${DOTNET_SDK} AS publish
-COPY --from=assets /repo /repo
-WORKDIR /repo
-RUN dotnet restore DfE.FindInformationAcademiesTrusts
-RUN dotnet build DfE.FindInformationAcademiesTrusts -c Release
-RUN dotnet publish DfE.FindInformationAcademiesTrusts -c Release -o /app --no-build
+# Build the app using the dotnet SDK
+FROM "mcr.microsoft.com/dotnet/sdk:${DOTNET_VERSION}-azurelinux3.0" AS build
+WORKDIR /build
+COPY . /build
+RUN ["dotnet", "restore", "DfE.FindInformationAcademiesTrusts"]
+RUN ["dotnet", "build", "DfE.FindInformationAcademiesTrusts", "--no-restore"]
+RUN ["dotnet", "publish", "DfE.FindInformationAcademiesTrusts",  "--no-restore", "-c", "Release", "-o", "/app"]
```

- We should use `--no-build` to ensure that the step above isn't rerun. This implicitly sets the `--no-restore` flag
- We don't actually need the configuration flag `-c` here as we shouldn't be running a build
```suggestion
RUN ["dotnet", "publish", "DfE.FindInformationAcademiesTrusts",  "--no-build", "--output /app"]
```